src/js/components/OptionalEnviromentComponent.jsx

```diff
@@ -14,6 +14,7 @@ var OptionalEnvironmentComponent = React.createClass({
 
   propTypes: {
     errorIndices: React.PropTypes.array,
+    generalError: React.PropTypes.string,
```

Comment:
Still, I would like to pass the `getErrorMessage` function to this component and let it do its own lookup via the AppModalComponent.